app/src/main/java/com/example/felipe/criminalintent/CrimeAdapter.kt

```diff
@@ -0,0 +1,89 @@
+package com.example.felipe.criminalintent
+
+import android.support.v4.app.FragmentActivity
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import kotlinx.android.synthetic.main.list_item_crime.view.*
+
+class CrimeAdapter(var crimes: List<Crime>, var activity: FragmentActivity?) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        val layoutInflater = LayoutInflater.from(activity)
+        return when (viewType) {
+            ViewType.POLICE.ordinal -> CrimeHolderPolice(layoutInflater, parent)
+            else -> CrimeHolder(layoutInflater, parent)
+        }
+    }
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        when (holder.itemViewType) {
```

Comment:
you can define a common `CrimeHolder` and extend it. If you do that, this code should be:
`(holder as? CrimeHolder)?.bind(crimes[position])`